src/MetricExporter.ts

```diff
@@ -82,49 +90,116 @@ export class DynatraceMetricExporter implements MetricExporter {
 			return;
 		}
 
-		const payload = serializeMetrics(metrics, this._userTags, this._prefix);
+		const dtMetrics = metrics.map((metric) => {
+			const dimensions = Object.entries(metric.labels).map(([key, value]) => ({ key, value }));
+			switch (metric.aggregator.kind) {
+				case AggregatorKind.SUM: {
+					const data = metric.aggregator.toPoint();
+					const normalizedMetric = this._dtMetricFactory
+						.createTotalCounter(
+							metric.descriptor.name,
+							dimensions,
+							data.value,
+							hrTimeToMilliseconds(data.timestamp)
+						);
+					if (normalizedMetric == null) {
+						this._warnNormalizationFailure(metric.descriptor.name);
+					}
+					return normalizedMetric;
+				}
+				case AggregatorKind.HISTOGRAM: {
+					const data = metric.aggregator.toPoint();
+					const { sum, count } = data.value;
+
+					// TODO remove this hack which pretends all data points had the same value
+					const avg = sum / count;
+
+					const value: SummaryValue = {
+						min: avg,
+						max: avg,
+						sum,
+						count
+					};
+					const normalizedMetric = this._dtMetricFactory
+						.createSummary(
+							metric.descriptor.name,
+							dimensions,
+							value,
+							hrTimeToMilliseconds(data.timestamp)
+						);
+					if (normalizedMetric == null) {
+						this._warnNormalizationFailure(metric.descriptor.name);
+					}
+					return normalizedMetric;
+				}
+				case AggregatorKind.LAST_VALUE: {
+					const data = metric.aggregator.toPoint();
+					const normalizedMetric = this._dtMetricFactory
+						.createGauge(
+							metric.descriptor.name,
+							dimensions,
+							data.value,
+							hrTimeToMilliseconds(data.timestamp)
+						);
+					if (normalizedMetric == null) {
+						this._warnNormalizationFailure(metric.descriptor.name);
+					}
+					return normalizedMetric;
+				}
+			}
+		});
+
+		const lines = dtMetrics
+			.filter((m): m is Metric => m != null)
+			.slice(0, getPayloadLinesLimit())
+			.map(m => m.serialize());
 
-		if (!payload) {
+		if (lines.length === 0) {
 			process.nextTick(resultCallback, { code: ExportResultCode.SUCCESS });
 			return;
 		}
 
+		const payload = lines.join("\n");
```

Comment:
The endpoint only accepts 1000 lines at a time so you should make sure that no more than 1000 lines are sent per request.